htlcswitch/switch.go

```diff
@@ -385,12 +368,7 @@ func (s *Switch) SendHTLC(firstHop lnwire.ShortChannelID, paymentID uint64,
 		htlc:           htlc,
 	}
 
-	if err := s.forward(packet); err != nil {
-		s.removePendingPayment(paymentID)
-		return err
-	}
-
-	return nil
+	return s.forward(packet)
```

Comment:
When do pending payments need to be removed due to failures? Never now under the new assumptions?